src/test/java/com/google/sps/ChatWebSocketTest.java

```diff
@@ -0,0 +1,215 @@
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.websocket.Session;
+import javax.websocket.RemoteEndpoint;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import static org.junit.Assert.assertEquals;
+
+import org.json.JSONObject;
+
+import org.mockito.ArgumentCaptor;
+import static org.mockito.Mockito.*;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import com.google.cloud.datastore.Key;
+
+import com.google.sps.CapstoneAuth;
+import com.google.sps.DatastoreManager;
+import com.google.sps.ChatWebSocket;
+import com.google.sps.WebSocketHandler;
+import static com.google.sps.ChatWebSocket.*;
+
+@PrepareForTest({CapstoneAuth.class, DatastoreManager.class,
+    WebSocketHandler.class, Key.class})
+@RunWith(PowerMockRunner.class)
+public class ChatWebSocketTest {
+
+    public static final String TEST_ID_TOKEN = "1234testtoken";
+    public static final String TEST_UID = "testuid";
+    public static final String TEST_USER_NAME = "testname";
+    public static final String TEST_MESSAGE = "Hello World!";
+    public static final String TEST_TITLE = "TESTTITLE";
+    public static final String TEST_BODY = "Test Body";
+
+    private DatastoreManager mockManager;
+    private Session testSession;
+    private WebSocketHandler mockHandler;
+
+    @Before
+    public void setUp() {
+        mockManager = mock(DatastoreManager.class);
+        when(mockManager.isUserAllowedChatroom(TEST_UID, 1)).thenReturn(true);
+        when(mockManager.getUserName(TEST_UID)).thenReturn(TEST_USER_NAME);
+
+        mockHandler = mock(WebSocketHandler.class);
+
+        testSession = mock(Session.class);
+
+        PowerMockito.mockStatic(DatastoreManager.class);
+        when(DatastoreManager.getInstance()).thenReturn(mockManager);
+
+        PowerMockito.mockStatic(WebSocketHandler.class);
+        when(WebSocketHandler.getInstance()).thenReturn(mockHandler);
+    }
+
+    @Test
+    public void checkOnOpen() throws IOException {
+        Map<String, String> pathParam =
+            Collections.singletonMap(SOCKET_PARAMETER_ROOM, "1");
+        when(testSession.getPathParameters())
+            .thenReturn(pathParam);
+
+        Map<String, List<String>> requestParam
+            = Collections.singletonMap(SOCKET_PARAMETER_ID,
+                Arrays.asList(TEST_ID_TOKEN));
+        when(testSession.getRequestParameterMap()).thenReturn(requestParam);
+
+        PowerMockito.mockStatic(CapstoneAuth.class);
+        when(CapstoneAuth.isUserAuthenticated(TEST_ID_TOKEN)).thenReturn(true);
+        when(CapstoneAuth.getUserId(TEST_ID_TOKEN)).thenReturn(TEST_UID);
+
+        ChatWebSocket instance = new ChatWebSocket();
+        instance.onOpen(testSession);
+
+        verify(mockHandler, times(1)).addSession(1, testSession);
```

Comment:
Nice! I like how you used times(1) to verify the interaction with the mock and assure it's only interacted with once.